### ISSUE TITLE
[Feature request] Support purview query parameters

### ISSUE THREAD

**Issue by catalinaperalta** [Azure/typespec-azure#21]
The purview datamap service defines query parameters in the following format:
```
attribute key-value pair(s) can be provided in the following format: attr:<attrName>=<attrValue>
```

Link to the original swagger definition: https://github.com/Azure/azure-rest-api-specs/blob/c81f1fad9691c6d3a804711700a4224a686c08f1/specification/purview/data-plane/Azure.Analytics.Purview.DataMap/stable/2023-09-01/purviewdatamap.json#L595-L599

How can we represent this in TypeSpec?

**Comment by catalinaperalta** [comment]
I think a Record type could work here, but we'd need to account for how to specify the `attr:` prefix before each key-value pair. Later on this also needs support in DPG as well

**Comment by timotheeguerin** [comment]
```
@query("attr:foo") foo: string
```
is this what you are looking for?

**Comment by catalinaperalta** [comment]
No, the service team cannot list of the possible query parameter key values since some of them are specific to a customer's own models. We need some way to specify a dynamic key/value pair for query parameters. In your example above, foo is only one possible attribute, there could be many others.

**Comment by markcowl** [comment]
- bigger issue: allowing `Record<string>` for query parameters, headers, etc.

**Comment by timotheeguerin** [comment]
Issues: 
- https://github.com/microsoft/typespec/issues/2476
- https://github.com/microsoft/typespec/issues/2479

**Comment by markcowl** [comment]
Closing in favor of #2476 (this should be part of the consideration for that design)
